### PR TITLE
Expose VirtualDirectory to In Process mode.

### PR DESCRIPTION
**Pull request by jkotalik**
For https://github.com/aspnet/IISIntegration/issues/427

### CODE REVIEW

File: src/AspNetCore/Src/aspnetcoreconfig.cxx [hunk 1]
```diff
@@ -470,4 +481,76 @@ ASPNETCORE_CONFIG::Populate(
     }
 
     return hr;
+}
+
+HRESULT
+ASPNETCORE_CONFIG::SetupAppPath(
+    IHttpContext*            pHttpContext,
```
Comment (pan-wang) [inline src/AspNetCore/Src/aspnetcoreconfig.cxx]: it seems nobody references to pHttpContext as you passed in the siteconfigpath and resolved the path from it

**Comment by Tratcher** [comment]
Is this the same value as is passed to out-of-proc via env var?

**Comment by jkotalik** [comment]
I believe so. I guess we could just read the env variable?

**Comment by jkotalik** [comment]
Though IMO it makes no sense to be reading env variables in process, as we have direct access to the structs.

File: src/AspNetCore/Inc/aspnetcoreconfig.h [hunk 2]
```diff
@@ -222,6 +230,14 @@ class ASPNETCORE_CONFIG : IHttpStoredContext
         IHttpContext *pHttpContext
     );
 
+    HRESULT
+    SetupAppPath(
```
Comment (pan-wang) [inline src/AspNetCore/Inc/aspnetcoreconfig.h]: it seems no one uses this new method. you can delete it

**Review by pan-wang — APPROVED** [review]

**Pull request merged**